server/polar/order/tasks.py

```diff
@@ -235,6 +235,58 @@ async def process_dunning_order(order_id: uuid.UUID) -> None:
         await order_service.process_dunning_order(session, order)
 
 
+@actor(
+    actor_name="order.remove_backfilled_events",
+    priority=TaskPriority.LOW,
+    time_limit=3600_000,
+)
+async def remove_backfilled_events(
+    batch_size: int = settings.DATABASE_STREAM_YIELD_PER, rate_limit_delay: float = 0.1
+) -> None:
+    """
+    Remove all previously backfilled order events (events with backfilled=true flag).
+    Useful for cleaning up before re-running backfill with updated logic.
+
+    Args:
+        batch_size: The number of events to delete at a time (default 100)
+        rate_limit_delay: Seconds to wait between batch deletes (default: 0.1)
+    """
+    async with AsyncSessionMaker() as session:
+        total_deleted = 0
+
+        while True:
+            result = await session.execute(
+                select(Event.id)
+                .where(
+                    Event.name.in_(
+                        [SystemEvent.order_paid, SystemEvent.order_refunded]
+                    ),
+                    Event.user_metadata["backfilled"].as_boolean().is_(True),
+                )
+                .limit(batch_size)
+            )
+            event_ids = [row[0] for row in result.all()]
+
+            if not event_ids:
+                break
+
+            await session.execute(delete(Event).where(Event.id.in_(event_ids)))
```

Comment:
We could probably perform that in one query by putting the "select" as a subquery:

```py
delete(Event).where(
    Event.id.in_(
        select(Event.id) # ...
    )
)
```

Reply:
Thanks!